src/search/ran1.c

```
@@ -4,34 +4,65 @@ file contains function:
 #include "real.h"
-#include <stdlib.h>
+#include <stdint.h>
+
+/**
+ * @brief Internal state for the xorshift64* PRNG.
+ *
+ * This implementation uses the xorshift64* algorithm (Vigna 2016), which
+ * provides better statistical properties and portability compared to libc
+ * rand(). The state is thread-local to avoid issues in multi-threaded use.
+ */
+static uint64_t ran1_state = 0;
```

Clang (14.0.6)

[warning] variable 'ran1_state' is non-const and globally accessible, consider making it const

(cppcoreguidelines-avoid-non-const-global-variables)